ttnn/cpp/ttnn/operations/ccl/shared_with_host/hetergeneous_data_structs.hpp

```diff
@@ -118,7 +118,7 @@ inline void advance_worker_global_page_interleaved (
 
     offset_into_worker_slice++;
 
-    uint32_t flattened_offset_worker_slice = offset_worker_slice.x + (offset_worker_slice.y * tensor_shape.x);
+    uint32_t flattened_offset_worker_slice = offset_worker_slice.x + (offset_worker_slice.y * tensor_slice_shape.x);
```

Comment:
Note for future readers. This was a bug fix that @avoraTT worked through. The originally implementation was slightly off.